--- origsrc/tesseract-3.04.00/ccutil/ambigs.cpp	2015-07-11 09:53:12.000000000 +0200
+++ src/tesseract-3.04.00/ccutil/ambigs.cpp	2015-07-28 23:49:08.285967500 +0200
@@ -24,13 +24,13 @@
 #include "helpers.h"
 #include "universalambigs.h"
 
-#if defined _WIN32 || defined(__CYGWIN__)
+#if defined _WIN32 
 #ifndef __GNUC__
 #define strtok_r strtok_s
 #else
 #include "strtok_r.h"
 #endif  /* __GNUC__ */
-#endif  /* _WIN32 __CYGWIN__*/
+#endif  /* _WIN32 */
 
 namespace tesseract {
 
--- origsrc/tesseract-3.04.00/configure.ac	2015-07-11 09:53:12.000000000 +0200
+++ src/tesseract-3.04.00/configure.ac	2015-07-29 00:09:51.557732200 +0200
@@ -88,7 +88,7 @@ case "${host_os}" in
         ;;
     cygwin*)
         AM_CONDITIONAL(ADD_RT, false)
-        AM_CONDITIONAL(T_WIN, true)
+        AM_CONDITIONAL(T_WIN, false)
         AC_SUBST([AM_LDFLAGS], ['-Wl,-no-undefined -Wl,--as-needed'])
         ;;
     solaris*)
--- origsrc/tesseract-3.04.00/training/pango_font_info.cpp	2015-07-11 09:53:12.000000000 +0200
+++ src/tesseract-3.04.00/training/pango_font_info.cpp	2015-07-28 23:32:10.261768400 +0200
@@ -18,6 +18,7 @@
  **********************************************************************/
 
 // Include automatically generated configuration file if running autoconf.
+#define _GNU_SOURCE
 #ifdef HAVE_CONFIG_H
 #include "config_auto.h"
 #endif
