--- origsrc/tesseract-3.04.00/ccutil/ambigs.cpp	2015-07-11 09:53:12.000000000 +0200
+++ src/tesseract-3.04.00/ccutil/ambigs.cpp	2015-07-26 18:12:20.918294800 +0200
@@ -24,13 +24,13 @@
 #include "helpers.h"
 #include "universalambigs.h"
 
-#if defined _WIN32 || defined(__CYGWIN__)
+#if defined _WIN32 
 #ifndef __GNUC__
 #define strtok_r strtok_s
 #else
 #include "strtok_r.h"
 #endif  /* __GNUC__ */
-#endif  /* _WIN32 __CYGWIN__*/
+#endif  /* _WIN32  */
 
 namespace tesseract {
 
--- origsrc/tesseract-3.04.00/viewer/svutil.cpp	2015-07-11 09:53:12.000000000 +0200
+++ src/tesseract-3.04.00/viewer/svutil.cpp	2015-07-26 18:12:20.933894800 +0200
@@ -222,7 +222,7 @@ void SVNetwork::Flush() {
 // This will always return one line of char* (denoted by \n).
 char* SVNetwork::Receive() {
   char* result = NULL;
-#if defined(_WIN32) || defined(__CYGWIN__)
+#if defined(_WIN32) 
   if (has_content) { result = strtok (NULL, "\n"); }
 #else
   if (buffer_ptr_ != NULL) { result = strtok_r(NULL, "\n", &buffer_ptr_); }
