--- origsrc/catdoc-0.94.2/CYGWIN-PATCHES/catdocrc	1969-12-31 18:00:00.000000000 -0600
+++ CYGWIN-PATCHES/catdocrc	2012-11-05 10:25:28.437500000 -0600
@@ -0,0 +1,17 @@
+# source_charset should match your windows locale and is only 
+# used on non-Unicode docs - before Word 97.
+
+#charset_path=/usr/share/catdoc
+#map_path=/usr/share/catdoc
+
+# override source charset with -s
+source_charset=cp1252
+
+# on cygwin the output locale detection always returns US-ASCII so turn it off
+use_locale=no
+# override target charset with -d
+target_charset=8859-1
+unknown_char='?'
+
+#format=ascii
+#format=tex
--- origsrc/catdoc-0.94.2/CYGWIN-PATCHES/postinstall.sh	1969-12-31 18:00:00.000000000 -0600
+++ CYGWIN-PATCHES/postinstall.sh	2012-11-05 10:25:28.437500000 -0600
@@ -0,0 +1,3 @@
+#!/bin/sh
+test -e /etc/catdocrc || cp etc/defaults/etc/catdocrc /etc/catdocrc
+
--- origsrc/catdoc-0.94.2/CYGWIN-PATCHES/preremove.sh	1969-12-31 18:00:00.000000000 -0600
+++ CYGWIN-PATCHES/preremove.sh	2012-11-05 10:25:28.437500000 -0600
@@ -0,0 +1,4 @@
+#!/bin/sh
+if [ ! -e /etc/catdocrc ]; then
+    cmp -s /etc/catdocrc /etc/defaults/etc/catdocrc && rm /etc/catdocrc
+fi
