--- origsrc/hdf5-1.10.2/configure.ac	2018-03-29 21:18:10.000000000 +0200
+++ src/hdf5-1.10.2/configure.ac	2018-05-21 21:01:18.519000000 +0200
@@ -848,19 +848,6 @@ AC_SUBST([RUNPARALLEL])
 AC_SUBST([TESTPARALLEL])
 
 ## ----------------------------------------------------------------------
-## Disable shared libraries on CYGWIN. (LK - 04/16/15)
-## A number of tests run by "make check" fail on CYGWIN, so for HDF5 v1.8.15
-## we will change the default for shared libraries to disabled.
-
-
-case "`uname`" in
-  CYGWIN*)
-    enable_shared="no"
-    CHECK_WARN="Shared libraries are not currently supported on CYGWIN."
-    ;;
-esac
-
-## ----------------------------------------------------------------------
 ## Check if they would like the Java native interface (JNI) compiled
 ##
 
