--- origsrc/cfitsio/CMakeLists.txt	2014-08-19 23:23:17.000000000 +0200
+++ CMakeLists.txt	2015-05-03 16:24:05.061493100 +0200
@@ -6,14 +6,15 @@ OPTION (BUILD_SHARED_LIBS "Build Shared
 OPTION (USE_PTHREADS "Thread-safe build (using pthreads)" OFF)
 
 set(CMAKE_MODULE_PATH ${CMAKE_MODULE_PATH} "${CMAKE_SOURCE_DIR}")
-set (LIB_DESTINATION "${CMAKE_INSTALL_PREFIX}/lib${LIB_SUFFIX}")
-set (INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include/")
+set(BIN_DESTINATION "${CMAKE_INSTALL_PREFIX}/bin")
+set(LIB_DESTINATION "${CMAKE_INSTALL_PREFIX}/lib")
+set(INCLUDE_INSTALL_DIR "${CMAKE_INSTALL_PREFIX}/include/")
 
 # Define project version
 SET(${PROJECT_NAME}_MAJOR_VERSION 3)
 SET(${PROJECT_NAME}_MINOR_VERSION 36)
 SET(${PROJECT_NAME}_VERSION ${${PROJECT_NAME}_MAJOR_VERSION}.${${PROJECT_NAME}_MINOR_VERSION})
-SET(LIB_NAME cfitsio)
+SET(LIB_NAME "cfitsio")
 
 # Define
 IF(MSVC)
@@ -59,8 +60,6 @@ TARGET_LINK_LIBRARIES(${LIB_NAME} m ${PT
 )
 
 SET_TARGET_PROPERTIES(${LIB_NAME} PROPERTIES VERSION ${${PROJECT_NAME}_VERSION} SOVERSION ${${PROJECT_NAME}_MAJOR_VERSION})
-install(TARGETS ${LIB_NAME} LIBRARY DESTINATION ${LIB_DESTINATION})
-install(FILES ${H_FILES} DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Devel)
 
 ENABLE_TESTING()
 
@@ -72,11 +71,27 @@ ADD_EXECUTABLE(cookbook cookbook.c)
 TARGET_LINK_LIBRARIES(cookbook ${LIB_NAME})
 ADD_TEST(cookbook cookbook)
 
-ADD_EXECUTABLE(FPack fpack.c fpackutil.c)
-TARGET_LINK_LIBRARIES(FPack ${LIB_NAME})
+ADD_EXECUTABLE(fpack fpack.c fpackutil.c)
+TARGET_LINK_LIBRARIES(fpack ${LIB_NAME})
 
-ADD_EXECUTABLE(Funpack funpack.c fpackutil.c)
-TARGET_LINK_LIBRARIES(Funpack ${LIB_NAME})
+ADD_EXECUTABLE(funpack funpack.c fpackutil.c)
+TARGET_LINK_LIBRARIES(funpack ${LIB_NAME})
+
+ADD_EXECUTABLE(fitscopy fitscopy.c)
+TARGET_LINK_LIBRARIES(fitscopy ${LIB_NAME})
+
+ADD_EXECUTABLE(speed speed.c)
+TARGET_LINK_LIBRARIES(speed ${LIB_NAME})
+
+ADD_EXECUTABLE(imcopy imcopy.c)
+TARGET_LINK_LIBRARIES(imcopy ${LIB_NAME})
+
+install(TARGETS ${LIB_NAME} fitscopy funpack fpack cookbook speed imcopy
+        RUNTIME DESTINATION ${BIN_DESTINATION}
+        LIBRARY DESTINATION ${LIB_DESTINATION}
+        ARCHIVE DESTINATION ${LIB_DESTINATION})
+install(FILES ${H_FILES}
+        DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Devel)
 
 # To expands the command line arguments in Windows, see:
 # http://msdn.microsoft.com/en-us/library/8bch7bkk.aspx
