--- origsrc/fftw-3.3.4/tests/check.pl	2013-03-18 13:10:45.000000000 +0100
+++ tests/check.pl	2014-04-13 11:48:31.620691700 +0200
@@ -225,7 +225,7 @@ sub one_random_test {
     if ($vtype > 0 && $g > 1) {
 	my $v = int(1 + rand($g));
 	$sz = "${sz}*${v}" if ($vtype == 1);
-	$sz = "${sz}v${v}" if ($vtype == 2);
+	$sz = "${sz}*${v}" if ($vtype == 2);
     }
     if ($mpi) {
 	my $stype = int(rand(3));
--- origsrc/fftw-3.3.4/threads/Makefile.am	2013-03-18 13:10:45.000000000 +0100
+++ threads/Makefile.am	2014-04-13 11:48:31.640691700 +0200
@@ -25,7 +25,7 @@ libfftw3@PREC_SUFFIX@_threads_la_SOURCES
 threads.h dft-vrank-geq1.c ct.c rdft-vrank-geq1.c hc2hc.c		\
 vrank-geq1-rdft2.c f77api.c f77funcs.h
 libfftw3@PREC_SUFFIX@_threads_la_CFLAGS = $(AM_CFLAGS) $(PTHREAD_CFLAGS)
-libfftw3@PREC_SUFFIX@_threads_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
+libfftw3@PREC_SUFFIX@_threads_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ -no-undefined
 if !COMBINED_THREADS
 libfftw3@PREC_SUFFIX@_threads_la_LIBADD = ../libfftw3@PREC_SUFFIX@.la
 endif
@@ -34,7 +34,7 @@ libfftw3@PREC_SUFFIX@_omp_la_SOURCES = a
 threads.h dft-vrank-geq1.c ct.c rdft-vrank-geq1.c hc2hc.c	\
 vrank-geq1-rdft2.c f77api.c f77funcs.h
 libfftw3@PREC_SUFFIX@_omp_la_CFLAGS = $(AM_CFLAGS) $(OPENMP_CFLAGS)
-libfftw3@PREC_SUFFIX@_omp_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@
+libfftw3@PREC_SUFFIX@_omp_la_LDFLAGS = -version-info @SHARED_VERSION_INFO@ -no-undefined
 if !COMBINED_THREADS
 libfftw3@PREC_SUFFIX@_omp_la_LIBADD = ../libfftw3@PREC_SUFFIX@.la
 endif
