--- origsrc/netcdf-cxx4-4.3.1/cxx4/Makefile.am	2019-09-12 20:34:58.000000000 +0200
+++ src/netcdf-cxx4-4.3.1/cxx4/Makefile.am	2019-11-29 14:49:21.929609300 +0100
@@ -13,6 +13,8 @@ lib_LTLIBRARIES = libnetcdf_c++4.la
 # http://www.gnu.org/s/libtool/manual/html_node/Updating-version-info.html
 libnetcdf_c__4_la_LDFLAGS = -version-info 2:0:1 -no-undefined
 
+libnetcdf_c__4_la_LIBADD = -lnetcdf
+
 # These headers will be installed in the users header directory.
 include_HEADERS = netcdf ncAtt.h ncCheck.h ncDim.h ncException.h	\
 ncGroup.h ncOpaqueType.h ncVar.h ncVlenType.h ncCompoundType.h		\
--- origsrc/netcdf-cxx4-4.3.1/plugins/Makefile.am	2019-09-12 20:34:58.000000000 +0200
+++ src/netcdf-cxx4-4.3.1/plugins/Makefile.am	2019-11-29 15:09:25.382242000 +0100
@@ -20,9 +20,11 @@ lib_LTLIBRARIES = libh5bzip2.la
 
 libh5bzip2_la_SOURCES = ${HDF5PLUGINSRC}
 libh5bzip2_la_LDFLAGS = -module -avoid-version -shared -export-dynamic -no-undefined
+libh5bzip2_la_LIBADD = -lhdf5
 
 libmisc_la_SOURCES = H5Zmisc.c H5Zutil.c h5misc.h
 libmisc_la_LDFLAGS = -module -avoid-version -shared -export-dynamic -no-undefined -rpath ${abs_builddir}
+libmisc_la_LIBADD = -lhdf5
 
 
 EXTRA_DIST=${PLUGINSRC} ${BZIP2SRC} ${PLUGINHDRS} ${BZIP2HDRS} \
