--- origsrc/flint-2.4.5/configure	2015-04-22 23:28:02.857865800 +0200
+++ configure	2015-04-23 00:20:59.456006000 +0200
@@ -5,12 +5,12 @@
 # (C) 2012, William Hart, Jean-Pierre Flori, Thomas DuBuisson
 # (C) 2012, Jan Engelhardt
 
-PREFIX="/usr/local"
-GMP_DIR="/usr/local"
-MPFR_DIR="/usr/local"
-NTL_DIR="/usr/local"
-GC_DIR="/usr/local"
-BLAS_DIR="/usr/local"
+PREFIX="/usr/"
+GMP_DIR="/usr/"
+MPFR_DIR="/usr/"
+NTL_DIR="/usr/"
+GC_DIR="/usr/"
+BLAS_DIR="/usr/"
 WANT_NTL=0
 WANT_BLAS=0
 SHARED=1
@@ -251,7 +251,7 @@ if [ "$WANT_BLAS" = "1" ]; then
    fi
    EXTRA_INC_DIRS="${EXTRA_INC_DIRS} ${BLAS_INC_DIR}"
    EXTRA_LIB_DIRS="${EXTRA_LIB_DIRS} ${BLAS_LIB_DIR}"
-   EXTRA_LIBS="${EXTRA_LIBS} openblas"
+   EXTRA_LIBS="${EXTRA_LIBS} blas"
 fi
 CONFIG_BLAS="#define HAVE_BLAS ${WANT_BLAS}"
 
