--- origsrc/flint-2.7.0/Makefile.in	2021-01-12 22:47:54.878455900 +0100
+++ Makefile.in	2021-01-12 22:50:07.581546300 +0100
@@ -198,11 +198,12 @@ else
 endif
 
 install: library
+	mkdir -p "$(DESTDIR)$(PREFIX)/bin"
 	mkdir -p "$(DESTDIR)$(PREFIX)/$(LIBDIR)"
 	mkdir -p "$(DESTDIR)$(PREFIX)/include/flint"
 	$(AT)if [ "$(FLINT_SHARED)" -eq "1" ]; then \
-	   cp $(FLINT_LIB) "$(DESTDIR)$(PREFIX)/$(LIBDIR)"; \
-	   cp -a $(shell ls $(FLINT_LIBNAME)*) "$(DESTDIR)$(PREFIX)/$(LIBDIR)"; \
+	   cp $(FLINT_LIB) "$(DESTDIR)$(PREFIX)/bin"; \
+	   cp $(FLINT_IMPLIB) "$(DESTDIR)$(PREFIX)/$(LIBDIR)"; \
 	fi
 	$(AT)if [ "$(FLINT_STATIC)" -eq "1" ]; then \
 		cp libflint.a "$(DESTDIR)$(PREFIX)/$(LIBDIR)"; \
--- origsrc/flint-2.7.0/configure	2021-01-12 22:47:54.571467600 +0100
+++ configure	2021-01-12 22:54:31.780090200 +0100
@@ -467,7 +467,7 @@ if [ -z "$FLINT_LIB" ]; then
          EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -compatibility_version $FLINT_MAJOR.$FLINT_MINOR"
          EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -current_version $FLINT_MAJOR.$FLINT_MINOR.$FLINT_PATCH"
          ;;
-      CYGWIN* | MINGW*)
+      MINGW*)
          FLINT_LIBNAME="libflint.dll"
          FLINT_LIB="libflint-$FLINT_MAJOR.dll"
          EXTRA_SHARED_FLAGS="-static-libgcc"
@@ -476,6 +476,12 @@ if [ -z "$FLINT_LIB" ]; then
          EXTRA_SHARED_FLAGS="$EXTRA_SHARED_FLAGS -Wl,-soname,libflint-$FLINT_MAJOR.dll.$FLINT_MINOR.$FLINT_PATCH"
          FLINT_DLL=1
          ;;
+      CYGWIN*)
+         FLINT_LIBNAME="cygflint.dll"
+         FLINT_LIB="cygflint-$FLINT_MAJOR.dll"
+         FLINT_IMPLIB="libflint.dll.a"
+         EXTRA_SHARED_FLAGS="-shared -Wl,--export-all-symbols -Wl,-soname,libflint-$FLINT_MAJOR.dll.$FLINT_MINOR.$FLINT_PATCH -Wl,-rpath,$GMP_LIB_DIR -Wl,-rpath,$MPFR_LIB_DIR -Wl,--out-implib=$FLINT_IMPLIB"
+         FLINT_DLL=1;;
       *)
          FLINT_LIBNAME="libflint.so"
          FLINT_LIB="libflint.so.$FLINT_MAJOR.$FLINT_MINOR.$FLINT_PATCH"
@@ -786,6 +792,7 @@ echo "FLINT_SHARED=$SHARED" >> Makefile
 echo "FLINT_LIB=$FLINT_LIB" >> Makefile
 echo "FLINT_LIBNAME=$FLINT_LIBNAME" >> Makefile
 echo "OS=$OS" >> Makefile
+echo "FLINT_IMPLIB=$FLINT_IMPLIB" >> Makefile
 echo "FLINT_SOLIB=$FLINT_SOLIB" >> Makefile
 echo "FLINT_MAJOR=$FLINT_MAJOR" >> Makefile
 echo "EXEEXT=$EXEEXT" >> Makefile
