--- origsrc/flint-2.5.2/Makefile.in	2015-08-13 18:16:22.000000000 +0200
+++ Makefile.in	2017-12-31 09:36:31.042251600 +0100
@@ -197,11 +197,12 @@ else
 endif
 
 install: library
+	mkdir -p "$(DESTDIR)$(PREFIX)/bin"
 	mkdir -p "$(DESTDIR)$(PREFIX)/$(LIBDIR)"
 	mkdir -p "$(DESTDIR)$(PREFIX)/include/flint"
 	$(AT)if [ "$(FLINT_SHARED)" -eq "1" ]; then \
-	   cp $(FLINT_LIB) "$(DESTDIR)$(PREFIX)/$(LIBDIR)"; \
-	   cp -a $(shell ls $(FLINT_LIBNAME)*) "$(DESTDIR)$(PREFIX)/$(LIBDIR)"; \
+	   cp $(FLINT_LIB) "$(DESTDIR)$(PREFIX)/bin"; \
+	   cp $(FLINT_IMPLIB) "$(DESTDIR)$(PREFIX)/$(LIBDIR)"; \
 	fi
 	$(AT)if [ "$(FLINT_STATIC)" -eq "1" ]; then \
 		cp libflint.a "$(DESTDIR)$(PREFIX)/$(LIBDIR)"; \
--- origsrc/flint-2.5.2/configure	2015-08-13 18:16:22.000000000 +0200
+++ configure	2017-12-31 09:04:18.311705700 +0100
@@ -435,11 +435,17 @@ if [ -z "$FLINT_LIB" ]; then
          FLINT_LIBNAME="libflint.dylib"
          FLINT_LIB="libflint-$FLINT_MAJOR.$FLINT_MINOR.$FLINT_PATCH.dylib"
          EXTRA_SHARED_FLAGS="-install_name $PREFIX/lib/$FLINT_LIB -compatibility_version $FLINT_MAJOR.$FLINT_MINOR -current_version $FLINT_MAJOR.$FLINT_MINOR.$FLINT_PATCH -Wl,-rpath,$GMP_LIB_DIR -Wl,-rpath,$MPFR_LIB_DIR";;
-      CYGWIN* | MINGW*)
+      MINGW*)
          FLINT_LIBNAME="libflint.dll"
          FLINT_LIB="libflint-$FLINT_MAJOR.dll"
          EXTRA_SHARED_FLAGS="-static-libgcc -shared -Wl,--export-all-symbols -Wl,-soname,libflint-$FLINT_MAJOR.dll.$FLINT_MINOR.$FLINT_PATCH -Wl,-rpath,$GMP_LIB_DIR -Wl,-rpath,$MPFR_LIB_DIR"
          FLINT_DLL=1;;
+      CYGWIN*)
+         FLINT_LIBNAME="cygflint.dll"
+         FLINT_LIB="cygflint-$FLINT_MAJOR.dll"
+ 	 FLINT_IMPLIB="libflint.dll.a"
+         EXTRA_SHARED_FLAGS="-shared -Wl,--export-all-symbols -Wl,-soname,libflint-$FLINT_MAJOR.dll.$FLINT_MINOR.$FLINT_PATCH -Wl,-rpath,$GMP_LIB_DIR -Wl,-rpath,$MPFR_LIB_DIR -Wl,--out-implib=$FLINT_IMPLIB"
+         FLINT_DLL=1;;
       *)
 	 FLINT_LIBNAME="libflint.so"
          FLINT_LIB="libflint.so.$FLINT_MAJOR.$FLINT_MINOR.$FLINT_PATCH"
@@ -714,6 +720,7 @@ echo "FLINT_STATIC=$STATIC" >> Makefile
 echo "FLINT_SHARED=$SHARED" >> Makefile
 echo "FLINT_LIB=$FLINT_LIB" >> Makefile
 echo "FLINT_LIBNAME=$FLINT_LIBNAME" >> Makefile
+echo "FLINT_IMPLIB=$FLINT_IMPLIB" >> Makefile
 echo "FLINT_SOLIB=$FLINT_SOLIB" >> Makefile
 echo "EXEEXT=$EXEEXT" >> Makefile
 echo "PREFIX=$PREFIX" >> Makefile
