--- origsrc/flint-2.5.2/Makefile.in	2017-12-31 10:03:27.169835200 +0100
+++ Makefile.in	2017-12-31 11:00:09.246168300 +0100
@@ -152,7 +152,7 @@ tests: library test_helpers.o $(TESTS)
 	$(AT)$(foreach ext, $(EXTENSIONS), $(foreach dir, $(patsubst $(ext)/%.h, %, $(wildcard $(ext)/*.h)), mkdir -p build/$(dir)/test; BUILD_DIR=$(CURDIR)/build/$(dir); export BUILD_DIR; MOD_DIR=$(dir); export MOD_DIR; $(MAKE) -f $(CURDIR)/Makefile.subdirs -C $(ext)/$(dir) tests || exit $$?;))
 	mkdir -p build/interfaces/test
 	$(AT)if [ "$(WANT_NTL)" -eq "1" ]; then \
-		$(MAKE) build/interfaces/test/t-NTL-interface; \
+		$(MAKE) build/interfaces/test/t-NTL-interface$(EXEEXT); \
 	fi
 
 define test_mod
@@ -180,8 +180,8 @@ ifndef MOD
 	$(AT)$(foreach dir, $(BUILD_DIRS), mkdir -p build/$(dir)/test; BUILD_DIR=../build/$(dir); export BUILD_DIR; $(MAKE) -f ../Makefile.subdirs -C $(dir) check || exit $$?;)
 	mkdir -p build/interfaces/test
 	$(AT)if [ "$(WANT_NTL)" -eq "1" ]; then \
-		$(MAKE) build/interfaces/test/t-NTL-interface; \
-		build/interfaces/test/t-NTL-interface; \
+		$(MAKE) build/interfaces/test/t-NTL-interface$(EXEEXT); \
+		build/interfaces/test/t-NTL-interface$(EXEEXT); \
 	fi
 else
 	$(AT)$(foreach mod, $(MOD), $(call test_mod,$(mod)) || exit $$?;)
