--- origsrc/TeXmacs-1.99.4-src/configure.in	2015-12-03 09:29:48.000000000 +0100
+++ src/TeXmacs-1.99.4-src/configure.in	2015-12-18 14:47:02.889070600 +0100
@@ -1160,7 +1160,7 @@ case "$CONFIG_GUI" in
     QT)
         CONFIG_QT="Qt"
         CONFIG_GUI_DEFINE="QTTEXMACS"
-#        AC_DEFINE(QTTEXMACS, 1, [Enable experimental Qt port])
+        AC_DEFINE(QTTEXMACS, 1, [Enable experimental Qt port])
 	;;
 esac
 
--- origsrc/TeXmacs-1.99.4-src/plugins/r/src/tm_r.c	2014-03-03 12:44:12.000000000 +0100
+++ src/TeXmacs-1.99.4-src/plugins/r/src/tm_r.c	2016-03-02 00:25:02.026923500 +0100
@@ -12,6 +12,7 @@
 
 #include <stdio.h>
 #include <sys/select.h>
+#include "config.h"
 #if HAVE_PTY_H
 #include <pty.h>
 #endif
--- origsrc/TeXmacs-1.99.4-src/src/CMakeLists.txt	2014-03-03 12:44:15.000000000 +0100
+++ src/TeXmacs-1.99.4-src/src/CMakeLists.txt	2015-12-18 13:14:33.888372900 +0100
@@ -175,6 +175,7 @@ SET(TeXmacs_Base_SRCS
     ./System/Link/pipe_link.cpp
     ./System/Link/socket_link.cpp
     ./System/Link/socket_server.cpp
+    ./System/Link/socket_notifier.cpp
     ./System/Link/texmacs_client.cpp
     ./System/Link/texmacs_server.cpp
     ./System/Link/tm_link.cpp
--- origsrc/TeXmacs-1.99.4-src/src/System/Link/socket_notifier.cpp	2015-06-15 10:44:50.000000000 +0200
+++ src/TeXmacs-1.99.4-src/src/System/Link/socket_notifier.cpp	2016-03-02 00:09:13.687086400 +0100
@@ -8,7 +8,7 @@
 * It comes WITHOUT ANY WARRANTY WHATSOEVER. For details, see the file LICENSE
 * in the root directory or <http://www.gnu.org/licenses/gpl-3.0.html>.
 ******************************************************************************/
-#ifndef QTTEXMACS
+#if !(defined (QTTEXMACS) && (defined (__MINGW__) || defined (__MINGW32__) || defined (QTPIPES)))
 
 #include "config.h"
 
--- origsrc/TeXmacs-1.99.4-src/src/System/config.in	2014-03-03 12:44:14.000000000 +0100
+++ src/TeXmacs-1.99.4-src/src/System/config.in	2015-12-18 15:19:38.244301100 +0100
@@ -117,6 +117,9 @@
 /* Enabling Qt pipes */
 #undef QTPIPES
 
+/* Enable experimental Qt port */
+#undef QTTEXMACS
+
 /* The size of `void *', as computed by sizeof. */
 #undef SIZEOF_VOID_P
 
