--- origsrc/pstoedit-3.75/configure.ac	2019-12-08 13:29:30.000000000 +0100
+++ src/pstoedit-3.75/configure.ac	2021-05-21 20:21:54.779311700 +0200
@@ -313,7 +313,7 @@ if test "x${PSTOEDIT_GCC_OK}" = xno ; th
 else
       if test -n "$PKG_CONFIG"; then
         AC_MSG_CHECKING(ImageMagick flags through pkg-config)
-	PKG_CHECK_MODULES(LIBMAGICK, ImageMagick++, 
+	PKG_CHECK_MODULES(LIBMAGICK, Magick++, 
 		HAVE_LIBMAGICK=yes 
 		LIBMAGICK_CFLAGS="-DHAVE_MAGIC $LIBMAGICK_CFLAGS" 
 		LIBMAGICK_LDFLAGS="$LIBMAGICK_LIBS" , 
