--- origsrc/librsb-1.2.0-rc4/configure.ac	2016-05-11 14:12:16.000000000 +0200
+++ src/librsb-1.2.0-rc4/configure.ac	2016-07-23 13:11:47.841197600 +0200
@@ -15,7 +15,7 @@ define([librsb_ver_patch],[0])dnl
 define([librsb_ver_prers],[-rc4])dnl
 dnl	************************************************************************
 define([librsb_ver_string],librsb_ver_major.librsb_ver_minor.librsb_ver_patch)dnl
-define([librsb_librsbvernum],[librsb_ver_major"0"librsb_ver_minor"00"])dnl
+define([librsb_librsbvernum],["0"])dnl
 define([librsb_lib_string],librsb_librsbvernum)dnl
 define([librsb_release],librsb_ver_string[]librsb_ver_prers)dnl
 define([librsbversion],[ifelse(librsbsvnversion,[trunk],librsbsvnrevision,librsb_release)])dnl
