--- origsrc/qhull-7.3.2/CMakeLists.txt	2019-12-17 12:05:43.733756600 +0100
+++ src/qhull-7.3.2/CMakeLists.txt	2019-12-17 12:05:44.499744000 +0100
@@ -626,18 +626,18 @@ install(TARGETS ${qhull_TARGETS_INSTALL}
 include(CMakePackageConfigHelpers)
 
 write_basic_package_version_file(
-    "${CMAKE_CURRENT_BINARY_DIR}/Qhull/QhullConfigVersion.cmake"
+    "${CMAKE_CURRENT_BINARY_DIR}/Config/QhullConfigVersion.cmake"
     VERSION ${qhull_VERSION}
     COMPATIBILITY AnyNewerVersion
 )
 
 export(EXPORT QhullTargets
-  FILE "${CMAKE_CURRENT_BINARY_DIR}/Qhull/QhullTargets.cmake"
+  FILE "${CMAKE_CURRENT_BINARY_DIR}/Config/QhullTargets.cmake"
   NAMESPACE Qhull::
 )
 
 configure_file(${PROJECT_SOURCE_DIR}/build/config.cmake.in
-  "${CMAKE_CURRENT_BINARY_DIR}/Qhull/QhullConfig.cmake"
+  "${CMAKE_CURRENT_BINARY_DIR}/Config/QhullConfig.cmake"
   @ONLY
 )
 
@@ -652,8 +652,8 @@ install(EXPORT QhullTargets
 )
 install(
   FILES
-    "${CMAKE_CURRENT_BINARY_DIR}/Qhull/QhullConfig.cmake"
-    "${CMAKE_CURRENT_BINARY_DIR}/Qhull/QhullConfigVersion.cmake"
+    "${CMAKE_CURRENT_BINARY_DIR}/Config/QhullConfig.cmake"
+    "${CMAKE_CURRENT_BINARY_DIR}/Config/QhullConfigVersion.cmake"
   DESTINATION
     ${ConfigPackageLocation}
   COMPONENT
