--- origsrc/hwloc-1.11.1/tests/hwloc_backends.c	2015-10-06 19:06:45.000000000 +0200
+++ src/hwloc-1.11.1/tests/hwloc_backends.c	2015-10-20 22:58:49.735290700 +0200
@@ -15,7 +15,7 @@
 #include <errno.h>
 #include <assert.h>
 
-#ifdef HWLOC_WIN_SYS
+#if defined(HWLOC_WIN_SYS) && !defined(__CYGWIN__)
 #include <fcntl.h>
 #include <sys/stat.h>
 #include <sys/types.h>
--- origsrc/hwloc-1.11.1/utils/hwloc/hwloc-bind.c	2015-10-06 19:06:45.000000000 +0200
+++ src/hwloc-1.11.1/utils/hwloc/hwloc-bind.c	2015-10-20 22:42:44.594179900 +0200
@@ -17,7 +17,7 @@
 
 #include "misc.h"
 
-#ifdef HWLOC_WIN_SYS
+#if defined(HWLOC_WIN_SYS) && !defined(__CYGWIN__)
 #include <process.h>
 #define execvp(a,b) (int)_execvp((a), (const char * const *)(b))
 #endif
