--- origsrc/slang-pre2.2.3-104/autoconf/aclocal.m4	2010-11-17 10:42:04.000000000 +0100
+++ src/slang-pre2.2.3-104/autoconf/aclocal.m4	2010-11-19 10:22:39.977310000 +0100
@@ -26,11 +26,11 @@ AC_DEFUN([JD_INIT],     dnl#{{{
 [
 #These variable are initialized by JD init function
 CONFIG_DIR=`pwd`
-cd $srcdir
-if test "`pwd`" != "$CONFIG_DIR"
-then
-  AC_MSG_ERROR("This software does not support configuring from another directory.   See the INSTALL file")
-fi
+dnl# cd $srcdir
+dnl# if test "`pwd`" != "$CONFIG_DIR"
+dnl# then
+dnl#   AC_MSG_ERROR("This software does not support configuring from another directory.   See the INSTALL file")
+dnl# fi
 dnl# if test "X$PWD" != "X"
 dnl# then
 dnl#  CONFIG_DIR="$PWD"
