--- CYGWIN-PATCHES/postinstall.sh	1969-12-31 18:00:00.000000000 -0600
+++ CYGWIN-PATCHES/postinstall.sh	2006-09-27 22:08:48.000000000 -0500
@@ -0,0 +1,14 @@
+#!/bin/sh
+
+# Add /usr/share/lyx/fonts to /etc/fonts/local.conf
+# if it is not already there.
+if [ -f /etc/fonts/local.conf ]; then
+    grep -q /usr/share/lyx/fonts /etc/fonts/local.conf
+    if [ $? -ne 0 ]; then
+	sed 's/^<\/fontconfig>/<dir>\/usr\/share\/lyx\/fonts<\/dir>
+<\/fontconfig>/' /etc/fonts/local.conf > /etc/fonts/local.conf.tmp
+	mv -f /etc/fonts/local.conf.tmp /etc/fonts/local.conf
+	fc-cache /usr/share/lyx/fonts
+    fi
+fi
+    
