--- origsrc/irssi-1.0.0/src/perl/common/Makefile.PL.in	2017-01-03 14:36:46.000000000 +0100
+++ src/irssi-1.0.0/src/perl/common/Makefile.PL.in	2017-01-17 15:49:17.592619100 +0100
@@ -1,4 +1,4 @@
-use ExtUtils::MakeMaker;our $AM_DEFAULT_VERBOSITY='@AM_DEFAULT_VERBOSITY@';require "../Makefile_silent.pm";
+use ExtUtils::MakeMaker;our $AM_DEFAULT_VERBOSITY='@AM_DEFAULT_VERBOSITY@';require "@top_srcdir@/src/perl/Makefile_silent.pm";
 
 WriteMakefile('NAME' => 'Irssi',
               'LIBS' => '',
--- origsrc/irssi-1.0.0/src/perl/irc/Makefile.PL.in	2017-01-03 14:36:46.000000000 +0100
+++ src/irssi-1.0.0/src/perl/irc/Makefile.PL.in	2017-01-17 15:49:17.608219100 +0100
@@ -1,4 +1,4 @@
-use ExtUtils::MakeMaker;our $AM_DEFAULT_VERBOSITY='@AM_DEFAULT_VERBOSITY@';require "../Makefile_silent.pm";
+use ExtUtils::MakeMaker;our $AM_DEFAULT_VERBOSITY='@AM_DEFAULT_VERBOSITY@';require "@top_srcdir@/src/perl/Makefile_silent.pm";
 
 WriteMakefile('NAME' => 'Irssi::Irc',
               'LIBS' => '',
--- origsrc/irssi-1.0.0/src/perl/textui/Makefile.PL.in	2017-01-03 14:36:46.000000000 +0100
+++ src/irssi-1.0.0/src/perl/textui/Makefile.PL.in	2017-01-17 15:49:17.623819100 +0100
@@ -1,4 +1,4 @@
-use ExtUtils::MakeMaker;our $AM_DEFAULT_VERBOSITY='@AM_DEFAULT_VERBOSITY@';require "../Makefile_silent.pm";
+use ExtUtils::MakeMaker;our $AM_DEFAULT_VERBOSITY='@AM_DEFAULT_VERBOSITY@';require "@top_srcdir@/src/perl/Makefile_silent.pm";
 
 WriteMakefile('NAME' => 'Irssi::TextUI',
               'LIBS' => '',
--- origsrc/irssi-1.0.0/src/perl/ui/Makefile.PL.in	2017-01-03 14:36:46.000000000 +0100
+++ src/irssi-1.0.0/src/perl/ui/Makefile.PL.in	2017-01-17 15:49:17.623819100 +0100
@@ -1,4 +1,4 @@
-use ExtUtils::MakeMaker;our $AM_DEFAULT_VERBOSITY='@AM_DEFAULT_VERBOSITY@';require "../Makefile_silent.pm";
+use ExtUtils::MakeMaker;our $AM_DEFAULT_VERBOSITY='@AM_DEFAULT_VERBOSITY@';require "@top_srcdir@/src/perl/Makefile_silent.pm";
 
 WriteMakefile('NAME' => 'Irssi::UI',
               'LIBS' => '',
