--- origsrc/aspell-0.60.8/prog/compress.c	2019-10-08 02:15:21.000000000 +0200
+++ src/aspell-0.60.8/prog/compress.c	2019-12-26 11:09:37.002349000 +0100
@@ -20,7 +20,7 @@
 
 #include <stdio.h>
 
-#if defined(__CYGWIN__) || defined (_WIN32)
+#if defined (_WIN32)
 
 #  include <io.h>
 #  include <fcntl.h>
--- origsrc/aspell-0.60.8/prog/prezip.c	2019-10-08 02:15:21.000000000 +0200
+++ src/aspell-0.60.8/prog/prezip.c	2019-12-26 11:09:37.002349000 +0100
@@ -34,7 +34,7 @@
 #include <string.h>
 #include <assert.h>
 
-#if defined(__CYGWIN__) || defined (_WIN32)
+#if defined (_WIN32)
 
 #  include <io.h>
 #  include <fcntl.h>
