--- origsrc/pinentry-1.0.0/configure.ac	2016-11-22 02:00:11.000000000 -0600
+++ src/pinentry-1.0.0/configure.ac	2017-06-29 02:01:09.689734300 -0500
@@ -584,7 +584,7 @@ fi
 # Check whether we should build the W32 pinentry.  This is actually
 # the simplest check as we do this only for that platform.
 #
-pinentry_w32=no
+pinentry_w32=yes
 test $have_w32_system = yes && pinentry_w32=yes
 AM_CONDITIONAL(BUILD_PINENTRY_W32, test "$pinentry_w32" = "yes")
 
--- origsrc/pinentry-1.0.0/w32/main.c	2016-11-07 03:55:43.000000000 -0600
+++ src/pinentry-1.0.0/w32/main.c	2017-06-29 02:06:13.671512800 -0500
@@ -19,6 +19,7 @@
 #include <config.h>
 #include <stdio.h>
 #include <stdlib.h>
+#include <unistd.h>
 #if WINVER < 0x0403
 # define WINVER 0x0403  /* Required for SendInput.  */
 #endif
