--- origsrc/pinentry-1.0.0/pinentry-wrapper	2017-06-28 19:08:13.289846900 -0500
+++ src/pinentry-1.0.0/pinentry-wrapper	2017-06-28 21:34:29.174622400 -0500
@@ -37,6 +37,9 @@ if [ -n "$display" -a "$DISPLAY" != "$di
 elif [ -n "$KDE_FULL_SESSION" ]; then
     kde_running=1
     kde_ver="$KDE_SESSION_VERSION"
+elif [ -n "$LXQT_SESSION_CONFIG" ]; then
+    kde_running=1
+    kde_ver=5
 fi
 
 # Check for presence of xprop binary
