--- origsrc/pinentry-1.0.0/pinentry/pinentry-curses.c	2016-11-05 05:12:41.000000000 -0500
+++ src/pinentry-1.0.0/pinentry/pinentry-curses.c	2017-06-28 19:06:00.737978400 -0500
@@ -18,6 +18,7 @@
    Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA
    02111-1307, USA  */
 
+#define NCURSES_INTERNALS
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
@@ -48,6 +49,7 @@
 
 #include <assuan.h>
 
+#undef ttytype
 #include "pinentry.h"
 
 #if GPG_ERROR_VERSION_NUMBER < 0x011900 /* 1.25 */
