--- origsrc/netcdf-4.3.0/configure.ac	2013-05-09 18:15:14.000000000 +0200
+++ src/netcdf-4.3.0/configure.ac	2013-10-03 17:59:38.202366100 +0200
@@ -875,6 +875,18 @@ AH_BOTTOM([#include "ncconfigure.h"])
 # debugging purposes.
 # cp confdefs.h my_config.h
 
+AC_MSG_CHECKING([if libtool needs -no-undefined flag to build shared libraries])
+case "`uname`" in
+  CYGWIN*|MINGW*|AIX*)
+    ## Add in the -no-undefined flag to LDFLAGS for libtool.
+    AC_MSG_RESULT([yes])
+    LDFLAGS="$LDFLAGS -no-undefined"
+    ;;
+  *)
+    ## Don't add in anything.
+    AC_MSG_RESULT([no])
+    ;;
+esac
 
 AC_MSG_NOTICE([generating header files and makefiles])
 AC_CONFIG_FILES([Makefile
