--- origsrc/4ti2-1.6.6/configure.ac	2015-06-21 10:34:55.829504600 +0200
+++ src/4ti2-1.6.6/configure.ac	2015-06-21 09:04:35.361036500 +0200
@@ -62,6 +62,7 @@
 AM_CONDITIONAL([ENABLE_ZSOLVE], [ test x${enable_zsolve} == xyes ])
 AM_CONDITIONAL([ENABLE_FIBER], [ test x${enable_fiber} == xyes ])
 
+LT_INIT([disable-static win32-dll])
 ## Using the C compiler
 AC_PROG_CC
 gl_EARLY
@@ -107,7 +108,6 @@
 AC_DISABLE_SHARED
 # This triggers selection of some optimization flags (-g -O2), 
 # so it should appear _after_ the AX_CXX_MAXOPT, AX_CC_MAXOPT calls.
-AC_PROG_LIBTOOL
 
 # Checks for header files.
 #AC_HEADER_STDC
