--- origsrc/leptonica-1.72/src/environ.h	2015-04-17 05:27:08.000000000 +0200
+++ src/leptonica-1.72/src/environ.h	2015-12-11 18:45:44.097477300 +0100
@@ -165,13 +165,8 @@ typedef uintptr_t l_uintptr_t;
  * and writing on Unix with this translation:
  *        /tmp/xxx  -->   /tmp/leptonica/xxx
  * By default, leptonica is distributed for Unix without this translation
- * (except on Cygwin, which runs on Windows).
  */
-#if defined (__CYGWIN__)
-  #define  ADD_LEPTONICA_SUBDIR    1
-#else
   #define  ADD_LEPTONICA_SUBDIR    0
-#endif
 
 
 /*--------------------------------------------------------------------*
