--- origsrc/pure-ftpd-1.0.46/configure.ac	2017-04-24 16:39:24.000000000 +0200
+++ configure.ac	2017-05-06 11:44:34.759001600 +0200
@@ -40,7 +40,6 @@ AX_CHECK_COMPILE_FLAG([-fwrapv], [CFLAGS
 AX_CHECK_COMPILE_FLAG([-fno-strict-aliasing], [CFLAGS="$CFLAGS -fno-strict-aliasing"])
 AX_CHECK_COMPILE_FLAG([-fno-strict-overflow], [CFLAGS="$CFLAGS -fno-strict-overflow"])
 AS_IF([echo `(uname -s) 2>/dev/null` | $GREP "CYGWIN" > /dev/null], [
-  AX_CHECK_LINK_FLAG([-Wl,--dynamicbase], [LDFLAGS="$LDFLAGS -Wl,--dynamicbase"])
   AX_CHECK_LINK_FLAG([-Wl,--nxcompat], [LDFLAGS="$LDFLAGS -Wl,--nxcompat"])
 ], [
   AS_IF([test `(uname -s) 2>/dev/null` = "DragonFly"],
