--- origsrc/UDUNITS-2-2.2.20/configure.ac	2015-11-15 21:35:31.551115000 +0100
+++ src/UDUNITS-2-2.2.20/configure.ac	2015-11-15 21:31:21.396807000 +0100
@@ -125,6 +125,20 @@ AC_CHECK_FUNCS([floor memmove memset mod
 
 AC_PROG_LIBTOOL
 
+AC_MSG_CHECKING([if libtool needs -no-undefined flag to build shared libraries])
+case "${host}" in
+   *-*-cygwin* | *-*-mingw*|*-*-aix*)
+    ## Add in the -no-undefined flag to LDFLAGS for libtool.
+    AC_MSG_RESULT([yes])
+    NOUNDEFINED=" -no-undefined"
+    ;;
+  *)
+    ## Don't add in anything.
+    AC_MSG_RESULT([no])
+    ;;
+esac
+AC_SUBST([NOUNDEFINED])
+
 AC_CONFIG_FILES([Makefile
                  lib/Makefile
                  lib/xmlFailures/Makefile
--- origsrc/UDUNITS-2-2.2.20/lib/Makefile.am	2015-06-03 20:25:57.000000000 +0200
+++ src/UDUNITS-2-2.2.20/lib/Makefile.am	2015-11-15 21:32:59.861438800 +0100
@@ -33,7 +33,7 @@ info_TEXINFOS 	= udunits2lib.texi
 #pdf_DATA = udunits2lib.pdf
 AM_MAKEINFOFLAGS = -I $(top_srcdir)
 AM_MAKEINFOHTMLFLAGS = --no-split -I $(top_srcdir)
-libudunits2_la_LDFLAGS	= -version-number 0:1:0
+libudunits2_la_LDFLAGS	= -version-number 0:1:0 ${NOUNDEFINED}
 
 lex_prefix = ut
 LEX_OUTPUT_ROOT = lex.$(lex_prefix)
