--- origsrc/CUnit-2.1-2/CUnit/Sources/Makefile.am	2006-03-21 02:30:53.000000000 +0100
+++ CUnit/Sources/Makefile.am	2013-03-15 22:53:26.292975100 +0100
@@ -67,7 +67,7 @@ SUBDIRS = $(COMPILE_DIRS)
 lib_LTLIBRARIES = libcunit.la
 libcunit_la_SOURCES = 
 libcunit_la_LIBADD = $(OBJECT_FILES_SHARED)
-libcunit_la_LDFLAGS = -version-info @LIBTOOL_CURRENT@:@LIBTOOL_REVISION@:@LIBTOOL_AGE@
+libcunit_la_LDFLAGS = -version-info @LIBTOOL_CURRENT@:@LIBTOOL_REVISION@:@LIBTOOL_AGE@ -no-undefined
 
 
 if ENABLE_TEST
