--- origsrc/cddlib-0.94m/src/Makefile.am	2018-05-23 10:18:27.000000000 +0200
+++ src/Makefile.am	2021-01-15 22:05:46.114990100 +0100
@@ -6,13 +6,15 @@ fourier \
 projection \
 adjacency \
 allfaces \
+cddexec
+
+check_PROGRAMS= \
 testcdd1 \
 testshoot \
 testcdd2 \
 testlp1 \
 testlp2 \
-testlp3 \
-cddexec
+testlp3
 
 # Build these binaries but do not install them; these are sample applications
 # that are mostly interesting to developers.
--- origsrc/cddlib-0.94m/src/Makefile.gmp.am	2018-05-23 10:18:27.000000000 +0200
+++ src/Makefile.gmp.am	2021-01-15 22:07:48.190724400 +0100
@@ -42,32 +42,32 @@ allfaces_gmp_SOURCES = allfaces.c
 allfaces_gmp_LDADD = $(gmp_ldflags)
 allfaces_gmp_CPPFLAGS = $(gmp_cppflags)
 
-bin_PROGRAMS += testshoot_gmp
+check_PROGRAMS += testshoot_gmp
 testshoot_gmp_SOURCES = testshoot.c
 testshoot_gmp_LDADD = $(gmp_ldflags)
 testshoot_gmp_CPPFLAGS = $(gmp_cppflags)
 
-bin_PROGRAMS += testcdd1_gmp
+check_PROGRAMS += testcdd1_gmp
 testcdd1_gmp_SOURCES = testcdd1.c
 testcdd1_gmp_LDADD = $(gmp_ldflags)
 testcdd1_gmp_CPPFLAGS = $(gmp_cppflags)
 
-bin_PROGRAMS += testcdd2_gmp
+check_PROGRAMS += testcdd2_gmp
 testcdd2_gmp_SOURCES = testcdd2.c
 testcdd2_gmp_LDADD = $(gmp_ldflags)
 testcdd2_gmp_CPPFLAGS = $(gmp_cppflags)
 
-bin_PROGRAMS += testlp1_gmp
+check_PROGRAMS += testlp1_gmp
 testlp1_gmp_SOURCES = testlp1.c
 testlp1_gmp_LDADD = $(gmp_ldflags)
 testlp1_gmp_CPPFLAGS = $(gmp_cppflags)
 
-bin_PROGRAMS += testlp2_gmp
+check_PROGRAMS += testlp2_gmp
 testlp2_gmp_SOURCES = testlp2.c
 testlp2_gmp_LDADD = $(gmp_ldflags)
 testlp2_gmp_CPPFLAGS = $(gmp_cppflags)
 
-bin_PROGRAMS += testlp3_gmp
+check_PROGRAMS += testlp3_gmp
 testlp3_gmp_SOURCES = testlp3.c
 testlp3_gmp_LDADD = $(gmp_ldflags)
 testlp3_gmp_CPPFLAGS = $(gmp_cppflags)
