--- origsrc/cddlib-0.94m/lib-src/Makefile.am	2020-02-12 11:58:42.000000000 +0100
+++ lib-src/Makefile.am	2021-01-15 22:01:35.298394600 +0100
@@ -10,7 +10,7 @@ cddproj.c \
 setoper.c
 libcdd_la_CPPFLAGS = -UGMPRATIONAL
 
-AM_LDFLAGS = -version-info $(libcdd_version_info) $(CDD_LDFLAGS)
+libcdd_la_LDFLAGS = -version-info $(libcdd_version_info) $(CDD_LDFLAGS)
 
 pkginclude_HEADERS = \
 cdd.h \
--- origsrc/cddlib-0.94m/lib-src/Makefile.gmp.am	2020-12-08 15:38:03.000000000 +0100
+++ lib-src/Makefile.gmp.am	2021-01-15 22:03:29.385441400 +0100
@@ -9,7 +9,7 @@ cddtypes_f.h
 
 libcddgmp_la_SOURCES = $(libcdd_la_SOURCES)
 libcddgmp_la_CPPFLAGS = -DGMPRATIONAL
-libcddgmp_la_LDFLAGS = $(AM_LDFLAGS) -lgmp
+libcddgmp_la_LDFLAGS = $(AM_LDFLAGS) -lgmp -version-info $(libcdd_version_info) $(CDD_LDFLAGS)
 # do not ship generated source files
 nodist_libcddgmp_la_SOURCES = \
 cddcore_f.c \
