--- origsrc/libgeotiff-1.4.2/Makefile.am	2016-08-18 20:52:23.000000000 +0200
+++ Makefile.am	2016-09-20 05:44:35.958637000 +0200
@@ -18,7 +18,7 @@ if TIFF_IS_CONFIG
 TIFF_CFLAGS = @TIFF_INC@ -DHAVE_TIFF=1
 endif
 
-AM_CFLAGS = -I./libxtiff $(PROJ_CFLAGS) $(TIFF_CFLAGS) \
+AM_CFLAGS = -I.$(srcdir)/libxtiff $(PROJ_CFLAGS) $(TIFF_CFLAGS) \
 			-DCSV_DATA_DIR=\"$(datadir)/epsg_csv\"
 
 csvdir = $(datadir)/epsg_csv
@@ -110,7 +110,7 @@ if CSV_IS_CONFIG
                              csv/unit_of_measure.c
 endif
 
-libgeotiff_la_LDFLAGS = -version-info 3:2:1
+libgeotiff_la_LDFLAGS = -version-info 3:2:1 ${NOUNDEFINED}
 
 libgeotiff_la_LIBADD = libxtiff/libxtiff.la
 
--- origsrc/libgeotiff-1.4.2/bin/Makefile.am	2016-08-11 17:40:10.000000000 +0200
+++ bin/Makefile.am	2016-09-20 05:44:35.958637000 +0200
@@ -12,7 +12,7 @@ if TIFF_IS_CONFIG
 TIFF_CFLAGS = @TIFF_INC@ -DHAVE_TIFF=1
 endif
 
-AM_CFLAGS = -I../ -I../libxtiff $(TIFF_CFLAGS) 
+AM_CFLAGS = -I../ -I$(srcdir)/.. -I$(srcdir)/../libxtiff $(TIFF_CFLAGS) 
 
 LDADD = ../libgeotiff.la
 
--- origsrc/libgeotiff-1.4.2/configure.ac	2016-08-11 17:40:10.000000000 +0200
+++ configure.ac	2016-09-20 05:44:35.974237100 +0200
@@ -281,8 +281,8 @@ elif test "$PROJ_LIB" != "" -a "$PROJ_IN
   PROJ_INC=-I$PROJ_INC
 
   AC_SUBST(PROJ_INC,$PROJ_INC)
-  AC_DEFINE(HAVE_PROJECTS_H)
-  AC_DEFINE(HAVE_LIBPROJ)
+  AC_DEFINE([HAVE_PROJECTS_H],1,[Define if proj_api.h exists])
+  AC_DEFINE([HAVE_LIBPROJ],1,[Define if libproj is available])
   AC_SUBST([HAVE_LIBPROJ])
   PROJ_CONFIG="yes"
 
@@ -317,7 +317,21 @@ fi
 AM_CONDITIONAL([CSV_IS_CONFIG], [test ! x$CSV_CONFIG = xno])
 
 
-AC_ARG_ENABLE(towgs84, [  --disable-towgs84       Disable WGS84 parameters for binary compatibility with pre-1.4.1], AC_DEFINE(GEO_NORMALIZE_DISABLE_TOWGS84))
+AC_ARG_ENABLE(towgs84, [  --disable-towgs84       Disable WGS84 parameters for binary compatibility with pre-1.4.1], AC_DEFINE([GEO_NORMALIZE_DISABLE_TOWGS84],1,"Disable WGS84 parameters"))
+
+AC_MSG_CHECKING([if libtool needs -no-undefined flag to build shared libraries])
+case "${host}" in
+   *-*-cygwin* | *-*-mingw*|*-*-aix*)
+    ## Add in the -no-undefined flag to LDFLAGS for libtool.
+    AC_MSG_RESULT([yes])
+    NOUNDEFINED=" -no-undefined"
+    ;;
+  *)
+    ## Don't add in anything.
+    AC_MSG_RESULT([no])
+    ;;
+esac
+AC_SUBST([NOUNDEFINED])
 
 dnl #########################################################################
 dnl Doxygen settings
--- origsrc/libgeotiff-1.4.2/libxtiff/Makefile.am	2016-08-11 17:40:10.000000000 +0200
+++ libxtiff/Makefile.am	2016-09-20 05:44:36.270637600 +0200
@@ -4,7 +4,7 @@ if TIFF_IS_CONFIG
 TIFF_CFLAGS = @TIFF_INC@ -DHAVE_TIFF=1
 endif
 
-AM_CFLAGS = -I../ $(TIFF_CFLAGS) 
+AM_CFLAGS = -I../ -I$(srcdir)/.. $(TIFF_CFLAGS) 
 
 libxtiff_la_SOURCES = xtiff.c
 
