--- origsrc/libde265-1.0.5/libde265/Makefile.am	2020-01-03 13:00:51.000000000 +0100
+++ libde265/Makefile.am	2020-08-14 17:53:32.757751700 +0200
@@ -19,7 +19,7 @@ if HAVE_VISIBILITY
  libde265_la_CXXFLAGS += -DHAVE_VISIBILITY
 endif
 
-libde265_la_LDFLAGS = -version-info $(LIBDE265_CURRENT):$(LIBDE265_REVISION):$(LIBDE265_AGE)
+libde265_la_LDFLAGS = -no-undefined -version-info $(LIBDE265_CURRENT):$(LIBDE265_REVISION):$(LIBDE265_AGE)
 
 libde265_la_SOURCES = \
   acceleration.h \
