--- origsrc/ImageMagick-6.9.11-27/configure.ac	2020-08-10 02:15:26.000000000 +0200
+++ configure.ac	2020-08-13 16:29:29.226941000 +0200
@@ -1026,9 +1026,9 @@ AC_CHECK_SIZEOF([long double])
 # Obtain size of a 'unsigned long long' and define as
 # SIZEOF_UNSIGNED_LONG_LONG.  If 'unsigned long long' is not
 # supported then the value defined is zero.
-AC_CHECK_SIZEOF(unsigned long long)
+AC_CHECK_SIZEOF([unsigned long long])
 
-AC_CHECK_SIZEOF(void *)
+AC_CHECK_SIZEOF([void *])
 
 AC_MSG_CHECKING([whether our compiler supports __func__])
 AC_TRY_COMPILE([],
