--- origsrc/onig-6.9.0/src/Makefile.am	2018-09-03 01:46:37.000000000 +0200
+++ src/Makefile.am	2018-09-12 12:58:24.013329300 +0200
@@ -45,7 +45,7 @@ libonig_la_SOURCES = regint.h regparse.h
 	gb18030.c koi8_r.c cp1251.c \
 	onig_init.c
 
-libonig_la_LDFLAGS = -version-info $(LTVERSION)
+libonig_la_LDFLAGS = -version-info $(LTVERSION) -no-undefined
 
 EXTRA_DIST = koi8.c mktable.c \
 	unicode_fold_data.c unicode_property_data.c \
@@ -57,11 +57,6 @@ EXTRA_DIST = koi8.c mktable.c \
 	make_property.sh \
 	gperf_fold_key_conv.py gperf_unfold_key_conv.py
 
-dll:
-	$(CXX) -shared -Wl,--output-def,libonig.def -o libonig.dll *.o \
-	$(LIBS)
-	strip libonig.dll
-
 
 # character-types-table source generator
 mktable: mktable.c regenc.h
