--- origsrc/arpack-ng-3.4.0/configure.ac	2016-07-02 21:52:15.000000000 +0200
+++ src/arpack-ng-3.4.0/configure.ac	2016-07-29 10:46:38.566629000 +0200
@@ -37,7 +37,6 @@ AC_ARG_VAR(SYMBOLSUFFIX, [suffix to add
 
 if test x"$SYMBOLSUFFIX" != x""; then
   dnl Need to rely on non-F77 features
-  AC_FC_LINE_LENGTH(unlimited)
   FFLAGS="$FFLAGS $ac_cv_fc_line_length"
 
   AX_CHECK_COMPILE_FLAG(-cpp, FFLAGS="$FFLAGS -cpp",
