diff --git a/Modules/FindRuby.cmake b/Modules/FindRuby.cmake
index e5ea210..fa506d5 100644
--- a/Modules/FindRuby.cmake
+++ b/Modules/FindRuby.cmake
@@ -205,6 +205,7 @@ if( "${Ruby_FIND_VERSION_SHORT_NODOT}" GREATER 18  OR  "${_RUBY_VERSION_SHORT_NO
      NAMES ruby/config.h  config.h
      HINTS
      ${RUBY_HDR_DIR}/${RUBY_ARCH}
+     ${RUBY_HDR_DIR}
      ${RUBY_ARCH_DIR}
      ${RUBY_ARCHHDR_DIR}
      )
