--- origsrc/cmake-3.12.0/Modules/FindOpenGL.cmake	2018-07-17 13:29:00.000000000 +0200
+++ Modules/FindOpenGL.cmake	2018-08-09 06:08:41.549377900 +0200
@@ -129,14 +129,7 @@
   set(OPENGL_USE_${_COMPONENT} 1)
 endforeach()
 
-if (CYGWIN)
-  find_path(OPENGL_INCLUDE_DIR GL/gl.h )
-  list(APPEND _OpenGL_REQUIRED_VARS OPENGL_INCLUDE_DIR)
-
-  find_library(OPENGL_gl_LIBRARY opengl32 )
-  find_library(OPENGL_glu_LIBRARY glu32 )
-
-elseif (WIN32)
+if (WIN32)
 
   if(BORLAND)
     set (OPENGL_gl_LIBRARY import32 CACHE STRING "OpenGL library for win32")
