--- origsrc/cmake-3.14.5/Modules/FindTclsh.cmake	2019-05-31 15:31:25.000000000 +0200
+++ Modules/FindTclsh.cmake	2019-06-02 14:44:08.638927600 +0200
@@ -16,13 +16,6 @@
   TCLSH_FOUND = TRUE if tclsh has been found
   TCL_TCLSH = the path to the tclsh executable
 
-In cygwin, look for the cygwin version first.  Don't look for it later
-to avoid finding the cygwin version on a Win32 build.
-#]=======================================================================]
-
-if(CYGWIN)
-  find_program(TCL_TCLSH NAMES cygtclsh83 cygtclsh80)
-endif()
 
 get_filename_component(TK_WISH_PATH "${TK_WISH}" PATH)
 get_filename_component(TK_WISH_PATH_PARENT "${TK_WISH_PATH}" PATH)
