--- CYGWIN-PATCHES/postinstall.sh	1970-01-01 01:00:00.000000000 +0100
+++ CYGWIN-PATCHES/postinstall.sh	2013-01-26 14:16:47.246886300 +0100
@@ -0,0 +1,17 @@
+if [ ! -f /etc/openmpi-default-hostfile ]
+then
+    /usr/bin/mkdir -p /etc
+    /usr/bin/cp /etc/defaults/etc/openmpi-default-hostfile /etc/openmpi-default-hostfile
+fi
+
+if [ ! -f /etc/openmpi-mca-params.conf ]
+then
+    /usr/bin/mkdir -p /etc
+    /usr/bin/cp /etc/defaults/etc/openmpi-mca-params.conf /etc/openmpi-mca-params.conf
+fi
+
+if [ ! -f /etc/openmpi-totalview.tcl ]
+then
+    /usr/bin/mkdir -p /etc
+    /usr/bin/cp /etc/defaults/etc/openmpi-totalview.tcl /etc/openmpi-totalview.tcl
+fi
