--- configure.ac	2014-08-22 08:00:00.000000000 +0200
+++ configure.ac	2014-08-24 07:15:49.535477800 +0200
@@ -141,6 +141,19 @@ else
    AC_MSG_RESULT([no])
 fi
 
+AC_MSG_CHECKING([if libtool needs -no-undefined flag to build shared libraries])
+case "`uname`" in
+  CYGWIN*|MINGW*|AIX*)
+    ## Add in the -no-undefined flag to LDFLAGS for libtool.
+    AC_MSG_RESULT([yes])
+    LDFLAGS="$LDFLAGS -no-undefined"
+    ;;
+  *)
+    ## Don't add in anything.
+    AC_MSG_RESULT([no])
+    ;;
+esac
+
 AC_CONFIG_FILES(
    [src/Makefile examples/Makefile Makefile])
 AC_OUTPUT
