--- origsrc/R-2.12.2/src/library/tools/R/install.R	2011-02-15 17:05:01.000000000 -0600
+++ src/R-2.12.2/src/library/tools/R/install.R	2011-03-07 20:28:34.946411600 -0600
@@ -712,7 +712,7 @@
                         } else {
                             setwd(owd)
                             for(arch in archs) {
-                                if (arch == "R") {
+                                if (arch == "R" || arch == "R.exe") {
                                     ## top-level, so one arch without subdirs
                                     has_error <- run_shlib(pkg_name, srcs, instdir, "")
                                 } else {
