--- origsrc/R-2.7.1/configure.ac	2008-06-23 04:39:06.000000000 -0500
+++ src/R-2.7.1/configure.ac	2008-07-08 21:00:54.140625000 -0500
@@ -1557,6 +1559,9 @@
     # RLAPACK_LDFLAGS="${wl}-bE:\$(top_builddir)/etc/Rlapack.exp"
     # LAPACK_LDFLAGS="${wl}-bI:\$(R_HOME)/etc/Rlapack.exp"
     ;;
+  cygwin*)
+    dylib_ldflags="${SHLIB_LDFLAGS} ${wl}--out-implib ${wl}\$@.a"
+    ;;
   darwin*)
     DYLIB_EXT=".dylib"
     dylib_ldflags="${darwin_dylib_ldflags}"
--- origsrc/R-2.7.1/src/extra/blas/Makefile.in	2008-06-23 04:38:25.000000000 -0500
+++ src/R-2.7.1/src/extra/blas/Makefile.in	2008-07-08 22:43:12.828125000 -0500
@@ -38,8 +38,12 @@
 
 Rblas_install: $(Rblas_la)
 	@$(MKINSTALLDIRS) $(DESTDIR)$(Rexeclibdir)
-	@$(SHELL) $(top_srcdir)/tools/copy-if-change $(Rblas_la) \
-	  $(DESTDIR)$(Rexeclibdir)/$(Rblas_la)
+	@BUILD_CYGWIN_FALSE@ @$(SHELL) $(top_srcdir)/tools/copy-if-change $(Rblas_la) \
+	@BUILD_CYGWIN_FALSE@  $(DESTDIR)$(Rexeclibdir)/$(Rblas_la)
+	@BUILD_CYGWIN_TRUE@ @$(SHELL) $(top_srcdir)/tools/copy-if-change $(Rblas_la) \
+	@BUILD_CYGWIN_TRUE@  $(DESTDIR)$(bindir)/$(Rblas_la)
+	@BUILD_CYGWIN_TRUE@ @$(SHELL) $(top_srcdir)/tools/copy-if-change libRblas.dll.a \
+	@BUILD_CYGWIN_TRUE@  $(DESTDIR)$(Rexeclibdir)/libRblas.dll.a
 
 Rblas_install-strip: $(Rblas_la)
 	@$(MKINSTALLDIRS) $(DESTDIR)$(Rexeclibdir)
--- origsrc/R-2.7.1/src/main/Makefile.in	2008-06-23 04:38:56.000000000 -0500
+++ src/R-2.7.1/src/main/Makefile.in	2008-07-08 22:45:48.890625000 -0500
@@ -193,7 +193,9 @@
 	@$(SHELL) $(top_srcdir)/tools/copy-if-change $(R_binary) "$(DESTDIR)$(Rexecbindir2)/R"
 install-lib: installdirs
 	@$(MKINSTALLDIRS) "$(DESTDIR)$(Rexeclibdir)"
-	@$(SHELL) $(top_srcdir)/tools/copy-if-change $(libR_la) "$(DESTDIR)$(Rexeclibdir)/$(libR_la)"
+	@BUILD_CYGWIN_FALSE@ @$(SHELL) $(top_srcdir)/tools/copy-if-change $(libR_la) "$(DESTDIR)$(Rexeclibdir)/$(libR_la)"
+	@BUILD_CYGWIN_TRUE@ @$(SHELL) $(top_srcdir)/tools/copy-if-change $(libR_la) "$(DESTDIR)$(bindir)/$(libR_la)"
+	@BUILD_CYGWIN_TRUE@ @$(SHELL) $(top_srcdir)/tools/copy-if-change libR.dll.a "$(DESTDIR)$(Rexeclibdir)/libR.dll.a"
 install-static: installdirs
 	@$(MKINSTALLDIRS) "$(DESTDIR)$(Rexeclibdir)"
 	@$(SHELL) $(top_srcdir)/tools/copy-if-change libR.a "$(DESTDIR)$(Rexeclibdir)/libR.a"
--- origsrc/R-2.7.1/src/modules/lapack/Makefile.in	2008-06-23 04:38:59.000000000 -0500
+++ src/R-2.7.1/src/modules/lapack/Makefile.in	2008-07-08 22:51:57.765625000 -0500
@@ -87,7 +87,9 @@
 
 install: installdirs
 	@$(SHELL) $(top_srcdir)/tools/copy-if-change $(lapack_la) "$(DESTDIR)$(Rexecmodulesdir)/$(lapack_la)"
-@USE_EXTERNAL_LAPACK_FALSE@	@$(SHELL) $(top_srcdir)/tools/copy-if-change $(Rlapack_la) "$(DESTDIR)$(Rexeclibdir)/$(Rlapack_la)"
+@BUILD_CYGWIN_FALSE@@USE_EXTERNAL_LAPACK_FALSE@	@$(SHELL) $(top_srcdir)/tools/copy-if-change $(Rlapack_la) "$(DESTDIR)$(Rexeclibdir)/$(Rlapack_la)"
+@BUILD_CYGWIN_TRUE@@USE_EXTERNAL_LAPACK_FALSE@	@$(SHELL) $(top_srcdir)/tools/copy-if-change $(Rlapack_la) "$(DESTDIR)$(bindir)/$(Rlapack_la)"
+@BUILD_CYGWIN_TRUE@@USE_EXTERNAL_LAPACK_FALSE@	@$(SHELL) $(top_srcdir)/tools/copy-if-change libRlapack.dll.a "$(DESTDIR)$(Rexeclibdir)/libRlapack.dll.a"
 installdirs:
 	@$(MKINSTALLDIRS) "$(DESTDIR)$(Rexecmodulesdir)"
 install-strip-Rlapack:
--- origsrc/R-2.7.1/src/nmath/standalone/Makefile.in	2008-06-23 04:38:57.000000000 -0500
+++ src/R-2.7.1/src/nmath/standalone/Makefile.in	2008-07-08 22:42:37.625000000 -0500
@@ -123,7 +123,9 @@
 	  -L. -lRmath $(LIBM)
 
 install: installdirs install-header @WANT_R_FRAMEWORK_FALSE@ install-pc
-@WANT_R_FRAMEWORK_FALSE@	@!(test -f $(libRmath_la)) || $(SHELL) $(top_srcdir)/tools/copy-if-change $(libRmath_la) $(DESTDIR)$(libdir)/$(libRmath_la)
+@BUILD_CYGWIN_FALSE@@WANT_R_FRAMEWORK_FALSE@	@!(test -f $(libRmath_la)) || $(SHELL) $(top_srcdir)/tools/copy-if-change $(libRmath_la) $(DESTDIR)$(libdir)/$(libRmath_la)
+@BUILD_CYGWIN_TRUE@@WANT_R_FRAMEWORK_FALSE@	@!(test -f $(libRmath_la)) || $(SHELL) $(top_srcdir)/tools/copy-if-change $(libRmath_la) $(DESTDIR)$(bindir)/$(libRmath_la)
+@BUILD_CYGWIN_TRUE@@WANT_R_FRAMEWORK_FALSE@	@!(test -f libRmath.dll.a) || $(SHELL) $(top_srcdir)/tools/copy-if-change libRmath.dll.a $(DESTDIR)$(libdir)/libRmath.dll.a
 @WANT_R_FRAMEWORK_FALSE@	@!(test -f libRmath.a) || $(SHELL) $(top_srcdir)/tools/copy-if-change libRmath.a $(DESTDIR)$(libdir)/libRmath.a
 @WANT_R_FRAMEWORK_TRUE@	@!(test -f $(libRmath_la)) || cp $(libRmath_la) $(R_FRAMEWORK_DIR)/Versions/$(FW_VERSION)/Resources/lib/$(libRmath_la) && install_name_tool -id $(R_FRAMEWORK_DIR)/Versions/$(FW_VERSION)/Resources/lib/$(libRmath_la) $(R_FRAMEWORK_DIR)/Versions/$(FW_VERSION)/Resources/lib/$(libRmath_la)
 @WANT_R_FRAMEWORK_TRUE@	@!(test -f libRmath.a) || $(SHELL) $(top_srcdir)/tools/copy-if-change libRmath.a $(R_FRAMEWORK_DIR)/Versions/$(FW_VERSION)/Resources/lib/libRmath.a
