--- src/R-3.1.2/Makefile.in	2015-01-31 00:05:06.498786800 +0100
+++ origsrc/R-3.1.2/Makefile.in	2014-05-21 00:15:03.000000000 +0200
@@ -19,7 +19,7 @@
 SUBDIRS = m4 tools doc etc share src tests
 SUBDIRS_WITH_NO_BUILD = po
 
-all: Makefile Makeconf R docs recommended vignettes 
+all: Makefile Makeconf R docs recommended vignettes javaconf
 recommended: @USE_RECOMMENDED_PACKAGES_TRUE@ stamp-recommended
 
 Makefile: $(srcdir)/Makefile.in $(top_builddir)/config.status
