--- origsrc/R-3.1.2/configure.ac	2015-01-31 00:33:40.319095500 +0100
+++ src/R-3.1.2/configure.ac	2015-01-31 00:35:01.530209800 +0100
@@ -1324,7 +1324,6 @@ case "${host_os}" in
     SHLIB_EXT=".dll"
     dylib_undefined_allowed=no
     is_cygwin=yes
-    main_ldflags="${wl}--large-address-aware ${wl}--stack=0xA00000"
     ;;
   darwin*)
     darwin_pic="-fPIC"
--- origsrc/R-3.1.2/etc/Makeconf.in	2014-07-15 00:15:14.000000000 +0200
+++ src/R-3.1.2/etc/Makeconf.in	2015-01-31 00:35:01.540209800 +0100
@@ -124,7 +124,7 @@ ALL_OBJCFLAGS = $(PKG_OBJCFLAGS) $(CPICF
 ALL_OBJCXXFLAGS = $(PKG_OBJCXXFLAGS) $(CXXPICFLAGS) $(SHLIB_CXXFLAGS) $(OBJCXXFLAGS)
 ALL_FFLAGS = $(R_XTRA_FFLAGS) $(PKG_FFLAGS) $(FPICFLAGS) $(SHLIB_FFLAGS) $(FFLAGS)
 ## LIBR here as a couple of packages use this without SHLIB_LINK
-ALL_LIBS = $(PKG_LIBS) $(SHLIB_LIBADD) $(LIBR)@DYLIB_UNDEFINED_ALLOWED_FALSE@ $(LIBINTL)
+ALL_LIBS = $(PKG_LIBS) $(SHLIB_LIBADD) $(LIBR)@DYLIB_UNDEFINED_ALLOWED_FALSE@ $(LIBINTL) $(LIBS)
 
 .SUFFIXES:
 .SUFFIXES: .c .cc .cpp .d .f .f90 .f95 .m .mm .M .o
--- origsrc/R-3.1.2/src/library/parallel/src/fork.c	2014-08-23 00:15:03.000000000 +0200
+++ src/R-3.1.2/src/library/parallel/src/fork.c	2015-01-31 00:35:01.550209800 +0100
@@ -229,15 +229,16 @@ static void setup_sig_handler() {
 #else
 /* sigaction is not viable, so use the "dumb" way
    to clean up anything that comes our way */
-static void setup_sig_handler() {
-    signal(SIGCHLD, parent_sig_handler);
-}
 
 static void parent_sig_handler(int sig) {
     /* clean up when a child terminates */
     if (sig == SIGCHLD)
 	clean_zombies();
 }
+
+static void setup_sig_handler() {
+    signal(SIGCHLD, parent_sig_handler);
+}
 #endif
 
 /* from Defn.h */
--- origsrc/R-3.1.2/src/library/stats/src/cov.c	2013-11-28 00:05:08.000000000 +0100
+++ src/R-3.1.2/src/library/stats/src/cov.c	2015-01-31 00:35:01.560209800 +0100
@@ -23,7 +23,7 @@
 #endif
 
 #ifdef HAVE_LONG_DOUBLE
-# define SQRTL sqrtl
+# define SQRTL sqrt
 #else
 # define SQRTL sqrt
 #endif
--- origsrc/R-3.1.2/src/nmath/dnbeta.c	2013-03-06 00:02:41.000000000 +0100
+++ src/R-3.1.2/src/nmath/dnbeta.c	2015-01-31 00:35:01.570209800 +0100
@@ -119,7 +119,7 @@ double dnbeta(double x, double a, double
     } while (term > sum * eps);
 
 #ifdef HAVE_LONG_DOUBLE
-    return R_D_exp((double)(p_k + logl(sum)));
+    return R_D_exp((double)(p_k + log(sum)));
 #else
     return R_D_exp((double)(p_k + log(sum)));
 #endif
--- origsrc/R-3.1.2/src/nmath/pnbeta.c	2014-03-13 00:03:20.000000000 +0100
+++ src/R-3.1.2/src/nmath/pnbeta.c	2015-01-31 00:35:01.580209800 +0100
@@ -93,7 +93,7 @@ pnbeta2(double x, double o_x, double a,
     /* return R_DT_val(ans), but we want to warn about cancellation here */
     if (lower_tail)
 #ifdef HAVE_LONG_DOUBLE
-	return (double) (log_p ? logl(ans) : ans);
+	return (double) (log_p ? log(ans) : ans);
 #else
 	return log_p ? log(ans) : ans;
 #endif
--- origsrc/R-3.1.2/src/nmath/pnchisq.c	2014-04-23 00:15:04.000000000 +0200
+++ src/R-3.1.2/src/nmath/pnchisq.c	2015-01-31 00:35:01.590209900 +0100
@@ -30,9 +30,9 @@
  */
 
 #ifdef HAVE_LONG_DOUBLE
-# define EXP expl
-# define FABS fabsl
-# define LOG logl
+# define EXP exp
+# define FABS fabs
+# define LOG log
 #else
 # define EXP exp
 # define FABS fabs
