--- origsrc/R-3.6.0/configure.ac	2019-03-29 00:02:40.000000000 +0100
+++ configure.ac	2019-04-29 14:41:43.302830900 +0200
@@ -1223,6 +1223,15 @@
       shlib_cxxldflags="-shared ${shlib_cxxldflags}"
     fi 
     ;;
+  cygwin*)
+    ## All Windows code is PIC
+    cpicflags=
+    cxxpicflags=
+    fpicflags=
+    fcpicflags=
+    SHLIB_EXT=".dll"
+    dylib_undefined_allowed=no
+    ;;
   darwin*)
     darwin_pic="-fPIC"
     dylib_undefined_allowed=no
@@ -1471,7 +1480,7 @@
 : ${CPICFLAGS="${cpicflags}"}
 if test -z "${CPICFLAGS}"; then
   case "${host_os}" in
-    aix*|mingw*)
+    aix*|mingw*|cygwin*)
       ;;
     *)
       AC_MSG_WARN([I could not determine CPICFLAGS.])
@@ -1483,7 +1492,7 @@
 : ${FPICFLAGS="${fpicflags}"}
 if test -z "${FPICFLAGS}"; then
   case "${host_os}" in
-    aix*|mingw*)
+    aix*|mingw*|cygwin*)
       ;;
     *)
       AC_MSG_WARN([I could not determine FPICFLAGS.])
@@ -1495,7 +1504,7 @@
 : ${CXXPICFLAGS="${cxxpicflags}"}
 if test -n "${CXX}" -a -z "${CXXPICFLAGS}"; then
   case "${host_os}" in
-    aix*|mingw*)
+    aix*|mingw*|cygwin*)
       ;;
     *)
       warn_cxxpicflags="I could not determine CXXPICFLAGS."
@@ -1549,6 +1558,9 @@
 dnl     # RLAPACK_LDFLAGS="${wl}-bE:\$(top_builddir)/etc/Rlapack.exp"
 dnl     # LAPACK_LDFLAGS="${wl}-bI:\$(R_HOME)/etc/Rlapack.exp"
 dnl     ;;
+  cygwin*)
+    dylib_ldflags="${SHLIB_LDFLAGS} ${wl}--out-implib ${wl}\$@.a"
+    ;;
   darwin*)
     DYLIB_EXT=".dylib"
     dylib_ldflags="${darwin_dylib_ldflags}"
