--- origsrc/R-3.4.0/configure.ac	2017-04-19 23:36:15.000000000 +0200
+++ configure.ac	2017-04-22 10:09:06.766462900 +0200
@@ -1285,6 +1285,15 @@ case "${host_os}" in
       shlib_cxxldflags="-shared ${shlib_cxxldflags}"
     fi 
     ;;
+  cygwin*)
+    ## All Windows code is PIC
+    cpicflags=
+    cxxpicflags=
+    fpicflags=
+    fcpicflags=
+    SHLIB_EXT=".dll"
+    dylib_undefined_allowed=no
+    ;;
   darwin*)
     darwin_pic="-fPIC"
     dylib_undefined_allowed=no
@@ -1532,7 +1541,7 @@ R_SH_VAR_ADD(MAIN_LDFLAGS, [${main_ldfla
 : ${CPICFLAGS="${cpicflags}"}
 if test -z "${CPICFLAGS}"; then
   case "${host_os}" in
-    aix*|mingw*)
+    aix*|mingw*|cygwin*)
       ;;
     *)
       AC_MSG_WARN([I could not determine CPICFLAGS.])
@@ -1544,7 +1553,7 @@ fi
 : ${FPICFLAGS="${fpicflags}"}
 if test -z "${FPICFLAGS}"; then
   case "${host_os}" in
-    aix*|mingw*)
+    aix*|mingw*|cygwin*)
       ;;
     *)
       AC_MSG_WARN([I could not determine FPICFLAGS.])
@@ -1556,7 +1565,7 @@ fi
 : ${CXXPICFLAGS="${cxxpicflags}"}
 if test -n "${CXX}" -a -z "${CXXPICFLAGS}"; then
   case "${host_os}" in
-    aix*|mingw*)
+    aix*|mingw*|cygwin*)
       ;;
     *)
       warn_cxxpicflags="I could not determine CXXPICFLAGS."
@@ -1610,6 +1619,9 @@ case "${host_os}" in
     # RLAPACK_LDFLAGS="${wl}-bE:\$(top_builddir)/etc/Rlapack.exp"
     # LAPACK_LDFLAGS="${wl}-bI:\$(R_HOME)/etc/Rlapack.exp"
     ;;
+  cygwin*)
+    dylib_ldflags="${SHLIB_LDFLAGS} ${wl}--out-implib ${wl}\$@.a"
+    ;;
   darwin*)
     DYLIB_EXT=".dylib"
     dylib_ldflags="${darwin_dylib_ldflags}"
