--- origsrc/xdelta3-3.0.9/Makefile.am	2015-03-31 06:52:07.000000000 +0200
+++ src/xdelta3-3.0.9/Makefile.am	2015-07-05 17:31:27.438120300 +0200
@@ -40,7 +40,7 @@ xdelta3regtest_SOURCES = $(common_SOURCE
 WFLAGS = -Wall -Wshadow -fno-builtin -Wextra -Wsign-compare \
 	 -Wextra -Wno-unused-parameter -Wno-unused-function
 
-C_WFLAGS = $(WFLAGS) -pedantic -std=c99
+C_WFLAGS = $(WFLAGS) 
 CXX_WFLAGS = $(WFLAGS)
 
 common_CFLAGS = \
--- origsrc/xdelta3-3.0.9/xdelta3-main.h	2015-03-31 07:58:19.000000000 +0200
+++ src/xdelta3-3.0.9/xdelta3-main.h	2015-07-05 17:31:27.448120300 +0200
@@ -49,6 +49,7 @@
 
 /*********************************************************************/
 
+
 #ifndef XD3_POSIX
 #define XD3_POSIX 0
 #endif
@@ -106,11 +107,13 @@ xsnprintf_func (char *str, int n, const
 #if XD3_POSIX
 #include <unistd.h> /* close, read, write... */
 #include <sys/types.h>
+#include <sys/time.h>
 #include <fcntl.h>
 #endif
 
 #ifndef _WIN32
 #include <unistd.h> /* lots */
+#include <sys/types.h>
 #include <sys/time.h> /* gettimeofday() */
 #include <sys/stat.h> /* stat() and fstat() */
 #else
@@ -138,6 +141,11 @@ static STARTUPINFO winStartupInfo;
 #endif
 
 /**********************************************************************
+ temporary declaration patch
+ *********************************************************************/
+int gettimeofday(struct timeval *, void *);
+
+/**********************************************************************
  ENUMS and TYPES
  *********************************************************************/
 
--- origsrc/xdelta3-3.0.9/xdelta3.h	2015-03-31 07:58:19.000000000 +0200
+++ src/xdelta3-3.0.9/xdelta3.h	2015-07-05 17:31:27.468120300 +0200
@@ -40,6 +40,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <sys/types.h>
+#include <sys/time.h>
 
 /****************************************************************/
 
